WeakAuras/Prototypes.lua

```diff
@@ -10113,6 +10113,167 @@ Private.event_prototypes = {
     },
     automaticrequired = true
   },
+  ["Currency"] = {
+    type = "unit",
+    canHaveDuration = true,
+    events = {
+      ["events"] = {
+        "CHAT_MSG_CURRENCY",
+        "CURRENCY_DISPLAY_UPDATE",
+      }
+    },
+    force_events = "CHAT_MSG_CURRENCY",
+    name = WeakAuras.newFeatureString..L["Currency"],
+    init = function(trigger)
+      local ret = [=[
+        local currencyInfo = C_CurrencyInfo.GetCurrencyInfo(%d)
+        if not currencyInfo then
+          currencyInfo = C_CurrencyInfo.GetCurrencyInfo(1) --Currency Token Test Token 4
+          currencyInfo.iconFileID = "Interface\\Icons\\INV_Misc_QuestionMark" --We don't want the user to think their input was valid
+        end
+      ]=]
+        return ret:format(trigger.currencyId or 1)
+    end,
+    statesParameter = "one",
+    args = {
+      {
+        name = "currencyId",
+        type = "currency",
+        values = "discovered_currencies",
+        sorted = true,
+        required = true,
+        display = L["Currency"],
+        store = true,
+        test = "true",
+      },
+      {
+        name = "name",
+        init = "currencyInfo.name",
+        hidden = true,
+        store = true,
+        test = "true",
+      },
+      {
+        name = "value",
+        init = "currencyInfo.quantity",
+        type = "number",
+        display = L["Quantity"],
+        store = true,
+        test = "true",
+        conditionType = "number",
+      },
+      {
+        name = "total",
+        init = "currencyInfo.maxQuantity",
+        type = "number",
+        display = L["Maximum Quantity"],
+        store = true,
+        conditionType = "number",
+      },
+      {
+        name = "progressType",
+        init = "'static'",
+        hidden = true,
+        store = true,
+        test = "true",
+      },
+      {
+        name = "icon",
+        init = "currencyInfo.iconFileID",
+        store = true,
+        hidden = true,
+        test = "true",
+      },
+      {
+        name = "capped",
+        init = [[value > 0 and value >= total]],
```

Comment:
this should check if total is > 0 or it can be true for a currency that does not have a maximum quantity